src/lava/magma/core/model/py/ports.py

```diff
@@ -162,12 +334,62 @@ def send(self, data: int, idx: int):
 
 
 class VarPortCmd:
+    """Helper class to specify constants. Used for communication between
+    PyRefPorts and PyVarPorts."""
     GET = enum_to_np(0)
     SET = enum_to_np(1)
 
 
 class PyRefPort(AbstractPyPort):
-    """Python implementation of RefPort used within AbstractPyProcessModels."""
+    """Python implementation of RefPort used within AbstractPyProcessModels.
+
+    A PyRefPort is a Port connected to a VarPort of a variable Var of another
+    Process. It is used to get or set the value of the referenced Var across
+    Processes. A PyRefPort is connected via two CSP channels and corresponding
+    CSP ports to a PyVarPort. One channel is used to send data from the
+    PyRefPort to the PyVarPort and the other channel is used to receive data
+    from the PyVarPort. PyRefPorts can get the value of a referenced Var
+    (read()) or set the value of a referenced Var (write()).
+
```

Comment:
Great explanation, thanks! To make it fully obvious, I would add, as second sentence, the precise difference to In/OutPorts.